### PR TITLE
Oval improvements

### PR DESCRIPTION
**Pull request by BirjuVachhani**

### CODE REVIEW

File: packages/flutter/lib/src/painting/circle_border.dart [hunk 1]
```diff
@@ -80,37 +79,25 @@ class CircleBorder extends OutlinedBorder {
...
   Path getOuterPath(Rect rect, { TextDirection? textDirection }) {
-    if (ovalness != 0) {
       return Path()..addOval(_adjustRect(rect));
-    }
-
-    return Path()
-      ..addOval(Rect.fromCircle(
-        center: rect.center,
-        radius: rect.shortestSide / 2.0,
-      ));
```
Comment (bernaferrari) [inline packages/flutter/lib/src/painting/circle_border.dart:113]: Why did you remove it?

File: packages/flutter/lib/src/painting/oval_border.dart [hunk 2]
```diff
@@ -22,6 +25,25 @@ class OvalBorder extends CircleBorder {
   /// Create an oval border.
   const OvalBorder({ super.side, super.ovalness = 1.0});
 
+  @override
+  ShapeBorder scale(double t) => OvalBorder(side: side.scale(t), ovalness: ovalness);
```
Comment (bernaferrari) [inline packages/flutter/lib/src/painting/oval_border.dart:29]: It is the same as circle border

File: packages/flutter/lib/src/painting/oval_border.dart [hunk 3]
```diff
@@ -22,6 +25,25 @@ class OvalBorder extends CircleBorder {
...
+  @override
+  ShapeBorder? lerpFrom(ShapeBorder? a, double t) {
+    if (a is CircleBorder) {
+      return OvalBorder(
+        side: BorderSide.lerp(a.side, side, t),
+        ovalness: ui.lerpDouble(a.ovalness, ovalness, t)!,
+      );
+    }
+    return super.lerpFrom(a, t);
+  }
```
Comment (bernaferrari) [inline packages/flutter/lib/src/painting/oval_border.dart:45]: Not doing this also works

File: packages/flutter/lib/src/painting/circle_border.dart (hunk 1 above)
Comment (BirjuVachhani) [reply packages/flutter/lib/src/painting/circle_border.dart:113]: Because I made it so the _adjustRect() can return value because it is already using that condition internally!!

File: packages/flutter/lib/src/painting/oval_border.dart (hunk 2 above)
Comment (BirjuVachhani) [reply packages/flutter/lib/src/painting/oval_border.dart:29]: Yes but if you don't override them here and then run Oval border tests, it fails because of mismatched object type.

File: packages/flutter/lib/src/painting/oval_border.dart (hunk 3 above)
Comment (BirjuVachhani) [reply packages/flutter/lib/src/painting/oval_border.dart:45]: Yeah, but the tests fails because those not overridden methods return CircleBorder because you're only overriding toString which has different representation.
